src/gc-arena/src/context.rs

```diff
@@ -211,7 +251,18 @@ impl Context {
                             work_done += sweep_size as f64;
                             self.allocation_debt
                                 .set((self.allocation_debt.get() - sweep_size as f64).max(0.0));
-                            Box::from_raw(sweep_ptr.as_ptr());
+                            if sweep.flags.alive() {
+                                // If the alive flag is set, that means we havn't dropped the inner value of this object,
+                                // so it is safe to drop normally.
+                                drop(Box::from_raw(sweep_ptr.as_ptr()));
+                            } else {
+                                // If the alive flag is not set, then that means we have already dropped the inner value,
+                                // so we only need to free the GcBox.
+                                alloc::alloc::dealloc(
+                                    sweep_ptr.as_ptr().cast(),
+                                    Layout::for_value(sweep),
```

Comment:
I *really* don't like doing things this way, I'd much rather have changed the value to be a `ManuallyDrop`.

Reply:
I've changed this to use `ManuallyDrop`. This now requires that we create a mutable reference in order to call `ManuallyDrop::drop` - though it does end up requiring a bit more code.